src/presenter/film-list-presenter.js

```diff
@@ -1,34 +1,86 @@
 import { render } from '../render';
-import { getContainerTemplates, getFilmSectionTemplates } from './container-template';
+import { getFilmSlice } from '../config';
+import { getContainerTemplates, getFilmSectionTemplates, getFilmListHeaders } from './container-template';
 import ContainerView from '../view/container-view';
-import FilmListView from '../view/film-list-view';
+import FilmListHeaderView from '../view/film-list-header-view';
 import FilmCardView from '../view/film-card-view';
 import ShowMoreView from '../view/show-more-view';
 
 import FilmPopupPresenter from './film-popup-presenter';
 
+//todo - далее перенести на уровень view
 const popupRootElement = document.querySelector('body');
 const getOpenPopup = () => popupRootElement.querySelector('.film-details');
+const filmSliceCount = getFilmSlice();
 
+//todo - единый÷ класс для 3х списков, но разные параметры создания экземпляра.
 export default class FilmListPresenter {
   #siteContainers = getContainerTemplates();
   #filmSectionTemplates = getFilmSectionTemplates();
-  // todo - перенести mainListSectionComponent и mainListContainerComponent в инит - если в dataset 0 записей - меняем заголовок (view) и контейнер mainListContainerComponent не формируем
-  #mainListSectionComponent = new FilmListView(this.#filmSectionTemplates.mainList);
+  #mainListSectionComponent = new ContainerView(this.#siteContainers.filmListSection);
   #mainListContainerComponent = new ContainerView(this.#siteContainers.filmListContainer);
+  #mainListHeaderComponent = new FilmListHeaderView();
+  #showMoreComponent = new ShowMoreView();
+  #renderStartIndex = 0;
+  #renderFilmCount = filmSliceCount;
+  #filmsData = [];
+  #filmListHeaders = getFilmListHeaders();
+  #filmListContext = null;
 
   init = (contentSection, dataSet, referenceDataModel) => {
     this.contentSection = contentSection;
-    this.dataSet = dataSet;
+    this.#filmsData = dataSet;
     this.referenceDataModel = referenceDataModel;
+    // todo - пока константа. далее надо как-то получать сведения о выбранном фильтре
+    // todo - текст заголовка перенести в конструктор вьюхи
+    this.#filmListContext = this.#filmListHeaders.ALLEMPTY;
+
 
     render(this.#mainListSectionComponent, this.contentSection);
+
+    if (this.#filmsData.length === 0) {
+      this.#renderHeader(this.#mainListHeaderComponent, this.#filmListContext, this.#mainListSectionComponent.element);
+      return;
+    }
+
+    this.#renderHeader(this.#mainListHeaderComponent, this.#filmListHeaders.NOEMPTY, this.#mainListSectionComponent.element, 'visually-hidden');
+    this.#renderFilmsContainer(this.#filmsData);
+
+  };
+
+  // #renderFilmsSectionComponent = (dataSet, target) => { };
+
+  #renderHeader = (component, text, target, cssClass = null) => {
+    component.element.textContent = text;
+    component.element.classList.add(cssClass);
```

Comment:
вот это место, например. могло бы выглядеть так
```suggestion
    component.setDecoratedText ({text,cssClass});
```
таким api со стороны view-шки ты подсказываешь коллегам, что два куска информации - text (строка) и cssClass - из известных имен в стилях - для целей отображения пользователю - работают в "транзакции".

ну конечно, в самом view эти два оператора все равно присутствуют - это да. Но они там к месту
а от сюда их надо "изолировать."